silkworm/rpc/http/connection.cpp

```diff
@@ -94,6 +97,21 @@ Task<void> Connection::do_read() {
     request_http_version_ = parser.get().version();
 
     if (boost::beast::websocket::is_upgrade(parser.get())) {
+        // Now that talking to the socket is succcessful,
+        // we tie the socket object to a websocket stream
+        boost::beast::websocket::stream<boost::beast::tcp_stream> stream(std::move(socket_));
+
+        auto websocket_connection = std::make_shared<WebSocketConnection>(std::move(stream), api_, std::move(handler_table_));
+        co_await websocket_connection->do_accept(parser.release());
+
+        SILK_TRACE << "Connection::run starting connection for websocket: " << &websocket_connection->ws();
+
+        auto connection_loop = [=]() -> Task<void> { co_await websocket_connection->read_loop(); };
+
+        boost::asio::co_spawn(io_context_, connection_loop, [&](const std::exception_ptr& eptr) {
+            if (eptr) std::rethrow_exception(eptr);
+        });
```

Comment:
This is the same we already do in `http::Server`. Not sure it's the best thing to handle a clean close in case of unexpected error and above all it will cause server termination, so it deserves some testing and a proper solution.